# https://github.com/r-spatial/sf/issues/1184
--- configure.ac
+++ configure.ac
@@ -232,9 +232,9 @@
   [--enable-proj], [enables PROJ.4 inclusion [default=yes]])],
   [], [enable_proj=yes])
 if test x"$enable_proj" != "xno"; then
-  AC_CHECK_HEADERS(proj.h, [proj_new_incl=1], 
-	[AC_CHECK_HEADERS(proj_api.h, [proj_new_incl=0], [AC_MSG_ERROR([cannot find proj_api.h, bailing out])])])
-  AC_SEARCH_LIBS(proj_normalize_for_visualization, proj, [proj_new_lib=1],
+  AC_SEARCH_LIBS(proj_normalize_for_visualization, proj, [proj_new_lib=1]
+  	AC_DEFINE(ACCEPT_USE_OF_DEPRECATED_PROJ_API_H)
+  	AC_CHECK_HEADER(proj_api.h,AC_DEFINE(HAVE_PROJ_API_H), [AC_MSG_ERROR([cannot find proj_api.h, bailing out])])
 	[AC_SEARCH_LIBS(pj_init_plus, proj, [proj_new_lib=0], [AC_MSG_ERROR(['libproj' is required but it doesn't seem to be installed on this system.])], [-lm -lpthread -lsqlite3])])
   if test $proj_new_incl -eq 1 && test $proj_new_lib -eq 1; then
     AC_DEFINE(PROJ_NEW)

--- config.h.in
+++ config.h.in
@@ -88,6 +88,9 @@
 /* Define to 1 if you have the `memset' function. */
 #undef HAVE_MEMSET
 
+/* Define to 1 to be able to use the deprecated <proj_api.h> header file. */
+#undef ACCEPT_USE_OF_DEPRECATED_PROJ_API_H
+
 /* Define to 1 if you have the <proj_api.h> header file. */
 #undef HAVE_PROJ_API_H
